sway-lsp/src/server.rs

```diff
@@ -131,20 +138,22 @@ impl LanguageServer for Backend {
     // Document Handlers
     async fn did_open(&self, params: DidOpenTextDocumentParams) {
         let uri = params.text_document.uri.clone();
+        let version = Some(params.text_document.version);
         let diagnostics = capabilities::text_sync::handle_open_file(self.session.clone(), &params);
-        self.publish_diagnostics(uri, diagnostics).await;
+        self.publish_diagnostics(uri, diagnostics, version).await;
     }
 
     async fn did_change(&self, params: DidChangeTextDocumentParams) {
         let uri = params.text_document.uri.clone();
+        let version = Some(params.text_document.version);
         let diagnostics = capabilities::text_sync::handle_change_file(self.session.clone(), params);
-        self.publish_diagnostics(uri, diagnostics).await;
+        self.publish_diagnostics(uri, diagnostics, version).await;
     }
 
     async fn did_save(&self, params: DidSaveTextDocumentParams) {
         let uri = params.text_document.uri.clone();
         let diagnostics = capabilities::text_sync::handle_save_file(self.session.clone(), &params);
-        self.publish_diagnostics(uri, diagnostics).await;
+        self.publish_diagnostics(uri, diagnostics, None).await;
```

Comment:
Is there a reason we don't know the `version` here? Is there any chance `DidSaveTextDocumentParams` should be updated to provide the version?

Reply:
There isn't a `version` within the `DidSaveTextDocumentParams` struct. I actually think it might be better to just stick with no version support like it was previously. Makes it simpler and less of our own state to manage and sync with the client. Will update that section now. We can address this in a separate PR if necessary.